src/coreclr/vm/codeversion.h

```diff
@@ -74,12 +74,14 @@ class NativeCodeVersion
     enum OptimizationTier
     {
         OptimizationTier0,
+        OptimizationTier0Instrumented,
```

Comment:
You may need to make changes similar to #66507, https://github.com/dotnet/diagnostics/pull/2928 and https://github.com/microsoft/perfview/pull/1584 to ensure this new tier and existing tieres still get reported properly by diagnostic tools.